### PR TITLE
Fix #386: Add support for local caching of audio & image assets

### LINKED ISSUE #386: Add temporary offline support for audio & image streaming
We need to include the audio assets for the supported fractions & ratio lessons with the APK binary and support local streaming rather than only remote streaming to be able to demonstrate offline support.

### PR DESCRIPTION
**Pull request by BenHenning**
Fix #386.

This PR introduces an AssetRepository that supports caching both JSON and and binary assets (for images and audio files). The repository caches JSON files in memory for faster retrieval, and caches binary files in the app's cache directory.

This PR also updates TopicListController to prime the repository with audio and image files from the 'meaning of equal parts' lesson. This will download all binary assets corresponding to that exploration upon opening the app. This takes a few minutes depending on the device's connection strength and consumes about ~80mb of cache space.

By default, the binary asset caching is disabled using a Dagger flag.

This PR does not introduce any tests because this local caching behavior is temporary until #169 is completed.

### CODE REVIEW

**Comment by BenHenning** [comment]
This is blocked on #393.

File: domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt [hunk 1]
```diff
@@ -14,10 +14,10 @@ import org.oppia.domain.util.StateRetriever
 import java.io.IOException
 import javax.inject.Inject
 
-const val TEST_EXPLORATION_ID_5 = "DIWZiVgs0km-"
-const val TEST_EXPLORATION_ID_6 = "test_exp_id_6"
-const val TEST_EXPLORATION_ID_30 = "30"
-const val TEST_EXPLORATION_ID_7 = "test_exp_id_7"
+const val TEST_EXPLORATION_ID_5 = "0"
+const val TEST_EXPLORATION_ID_6 = "1"
+const val TEST_EXPLORATION_ID_30 = "2"
```
Comment (vinitamurthi) [inline domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt:19]: Can we make the names of the variables same as the values? i.e. change TEST_EXPLORATION_ID_30 to TEST_EXPLORATION_ID_2?

**Review by vinitamurthi — APPROVED** [review]
I dont have much context on how it should work but code wise LGTM!

**Review by jamesxu0 — APPROVED** [review]
Really good implementation! I need to do a deeper look at AssetRepository to understand all functionality but in general LGTM.

**Comment by BenHenning** [comment]
Thanks!

File: domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt (hunk 1 above)
Comment (BenHenning) [reply domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt]: I think we should, but not in this PR. #370 is the first step toward this, and after that ID management should be much better.

Comment (BenHenning) [reply domain/src/main/java/org/oppia/domain/exploration/ExplorationRetriever.kt:19]: I think we should, but not in this PR. #370 is the first step toward this, and after that ID management should be much better.

**Pull request merged**